src/utils/mod.rs

```diff
@@ -19,6 +19,120 @@ macro_rules! good_panic {
     }}
 }
 
+pub fn make_lock_path(temp_path: &str, closure: &str) -> String {
+    let lock_hash =
+        &closure["/nix/store/".len()..closure.find("-").unwrap_or_else(|| closure.len())];
+    format!("{}/deploy-rs-canary-{}", temp_path, lock_hash)
+}
+
+fn make_emoji(level: log::Level) -> &'static str {
+    match level {
+        log::Level::Error => "❌",
+        log::Level::Warn => "⚠️",
+        log::Level::Info => "ℹ️",
+        log::Level::Debug => "❓",
+        log::Level::Trace => "🖊️",
+    }
+}
+
+pub fn logger_formatter_activate(
+    w: &mut dyn std::io::Write,
+    _now: &mut DeferredNow,
+    record: &Record,
+) -> Result<(), std::io::Error> {
+    let level = record.level();
+
+    write!(
+        w,
+        "⭐ {0} {1} {0} {2}",
+        make_emoji(level),
+        style(level, level.to_string()),
+        record.args()
```

Comment:
I don't know if flexi_logger supports fields or extended (per-process) metadata, but it would be good to get the PID of the process writing the log line here, to help me figure out https://github.com/serokell/deploy-rs/issues/38#issuecomment-753011354 (:

Reply:
That issue shouldn't exist anymore in this PR